lib/core/versions/latest/TransactionProcessor.ts

```diff
@@ -21,139 +21,122 @@ export default class TransactionProcessor implements ITransactionProcessor {
   public constructor (private downloadManager: DownloadManager, private operationStore: IOperationStore) { }
 
   public async processTransaction (transaction: TransactionModel): Promise<boolean> {
-    // Decode the anchor string
-    let anchoredData: AnchoredData;
-
     try {
-      anchoredData = AnchoredDataSerializer.deserialize(transaction.anchorString);
-    } catch (e) {
-      const message = `Anchored data deserialization failed for anchor string ${transaction.anchorString}. Error: ${e.message}`;
-      if (e instanceof SidetreeError) {
-        console.info(`Ignoring error: ${message}`);
-        return true;
-      } else {
-        console.error(`Unexpected error processing anchor string, MUST investigate and fix: ${message}`);
-        return false;
-      }
-    }
+      // Decode the anchor string.
+      const anchoredData = AnchoredDataSerializer.deserialize(transaction.anchorString);
 
-    try {
+      // Verify enough fee paid.
       FeeManager.verifyTransactionFeeAndThrowOnError(transaction.transactionFeePaid, anchoredData.numberOfOperations, transaction.normalizedTransactionFee);
-    } catch (e) {
-      const message = `Fee verification failure for anchor string ${transaction.anchorString}. Error: ${e.message}`;
-      if (e instanceof SidetreeError) {
-        console.info(`Ignoring error: ${message}`);
+
+      // Download and verify anchor file.
+      const anchorFile = await this.downloadAndVerifyAnchorFile(anchoredData.anchorFileHash);
+
+      // Download and verify batch file.
+      const operations = await this.downloadAndVerifyBatchFile(anchorFile, transaction);
+
+      // If the code reaches here, it means that the batch of operations is valid, process the operations.
+      const endTimer = timeSpan();
```

Comment:
Should we include this endTimer in the finally so that we always log the time taken? Maybe we can also log the success or failure ... maybe good for telemetry later. As in how much time are the success vs. bad transactions taking?

Reply:
Synced off line. Remove timing measurement until later.